### PR TITLE
feat(telemetry): Remove consent confirmation prompt for `kedro-telemetry`

### PR DESCRIPTION
**Pull request by ElenaKhaustova**
## Description
Partly solves https://github.com/kedro-org/kedro-plugins/issues/726

## Development notes

- `_confirm_consent` was removed
- now we use consent from the `.telemetry` file if it exists and has a valid format otherwise, we consider consent is `True`

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by astrojuanlu** [comment]
A few things:

- I contend that we should merge this one before proceeding with https://github.com/kedro-org/kedro/pull/3976
- Maybe we could try to address #728 in this same PR
- I gave this a try with a previously created project, and got this:

```
$ kedro run
[07/04/24 08:19:51] INFO     Using `conf/logging.yml` as logging configuration. You can change this by setting the KEDRO_LOGGING_CONFIG environment __init__.py:249
                             variable accordingly.                                                                                                                 
                    ERROR    Failed to retrieve UUID: badly formed hexadecimal UUID string                                                             plugin.py:78
[07/04/24 08:19:52] WARNING  Failed to send data to Heap. Response code returned: 400, Response reason: Bad Request                                   plugin.py:336
                    WARNING  Failed to send data to Heap. Response code returned: 400, Response reason: Bad Request                                   plugin.py:336
```

which is probably not related to this PR, but maybe it's too scary (specially given that it's printed twice, maybe because of #730?). I think there are 2 things here:

- The "Failed to retrieve UUID" message introduced in #596 could maybe be "Could not find or create project UUID" and maybe set it as a warning (although I'm hesitating whether the user needs to see it at all)
- The HTTP 400 to Heap was unexpected, maybe there's something wrong when the Project UUID is empty? https://github.com/kedro-org/kedro-plugins/blob/92bf6ebaa249c9e144fc6fee18a83b76973c13f2/kedro-telemetry/kedro_telemetry/plugin.py#L80

And finally, the "Kedro is sending anonymous usage data" message I proposed in https://github.com/kedro-org/kedro-plugins/issues/715 is not being shown.

**Comment by ElenaKhaustova** [comment]
> A few things:
> 
> * I contend that we should merge this one before proceeding with [Make `kedro-telemetry` a core dependency kedro#3976](https://github.com/kedro-org/kedro/pull/3976)
> * Maybe we could try to address [`kedro-telemetry`: Introduce environment variables for disabling telemetry and revise disabling methods #728](https://github.com/kedro-org/kedro-plugins/issues/728) in this same PR
> * I gave this a try with a previously created project, and got this:
> 
> ```
> $ kedro run
> [07/04/24 08:19:51] INFO     Using `conf/logging.yml` as logging configuration. You can change this by setting the KEDRO_LOGGING_CONFIG environment __init__.py:249
>                              variable accordingly.                                                                                                                 
>                     ERROR    Failed to retrieve UUID: badly formed hexadecimal UUID string                                                             plugin.py:78
> [07/04/24 08:19:52] WARNING  Failed to send data to Heap. Response code returned: 400, Response reason: Bad Request                                   plugin.py:336
>                     WARNING  Failed to send data to Heap. Response code returned: 400, Response reason: Bad Request                                   plugin.py:336
> ```
> 
> which is probably not related to this PR, but maybe it's too scary (specially given that it's printed twice, maybe because of #730?). I think there are 2 things here:
> 
> * The "Failed to retrieve UUID" message introduced in [feat(telemetry): Unique User IDs in kedro-telemetry - merge only for kedro-telemetry release 0.4.0 #596](https://github.com/kedro-org/kedro-plugins/pull/596) could maybe be "Could not find or create project UUID" and maybe set it as a warning (although I'm hesitating whether the user needs to see it at all)
> * The HTTP 400 to Heap was unexpected, maybe there's something wrong when the Project UUID is empty? https://github.com/kedro-org/kedro-plugins/blob/92bf6ebaa249c9e144fc6fee18a83b76973c13f2/kedro-telemetry/kedro_telemetry/plugin.py#L80
> 
> And finally, the "Kedro is sending anonymous usage data" message I proposed in #715 is not being shown.

@astrojuanlu, thank you for the review!

1. I believe there is not much difference if we merge this PR before #3976 or after as it only removes the consent confirmation prompt and works fine with the latest `kedro==0.19.6` version and changes introduced in the PR #3976
2. I think we can follow @DimedS split and address #727 ("Kedro is sending anonymous usage data" will be added with this PR) and #728 separately on top of this PR and, once they're ready, merge them together one by one.
3. I was not able to reproduce the errors you mentioned. I've tested the current change on `spaceflights-pandas` with `kedro==0.19.6` and changes from here #3976. Both work well. I double checked the response status when sending to the heap - it looks good - `200 OK`. I also checked that project ID is creating well and is part of the query. But anyway, the current changes cannot not affect the project ID logic. Could you please check if the error is not appearing for you without current changes?

**Review by DimedS — APPROVED** [review]
Thank you, @ElenaKhaustova. I tested all the PRs together, and everything works well. I also checked the telemetry info on Heap, and it's functioning correctly.

I believe we can merge the current PR and [this one](https://github.com/kedro-org/kedro/pull/3976), then continue with [message notification](https://github.com/kedro-org/kedro-plugins/issues/727) and [new disabling methods](https://github.com/kedro-org/kedro-plugins/issues/728), followed by other telemetry tickets.

I agree on merging the [Starters PR](https://github.com/kedro-org/kedro-starters/pull/224) last, just before the release.

File: kedro-telemetry/kedro_telemetry/plugin.py
Comment (merelcht) [inline kedro-telemetry/kedro_telemetry/plugin.py:350]:
```suggestion
    Use telemetry consent from ".telemetry" file if it exists and has a valid format.
```

**Review by merelcht — APPROVED** [review]
I didn't test this, but from a code perspective this looks good to me. I agree that the environment variables should be done separately. We just need to make sure we don't do any releases until the full opt-out flow is implemented in all parts.

**Pull request merged**
